### PR TITLE
feat(repository): Implement Create org app token functionality 

### PR DESCRIPTION
**Pull request by mikemountain**
## Description
This PR adds the functionality for creating an apptoken based off of a provided domain object in an org scope, inserting all the pertinent information into the db, as well as a small refactor to allow for dynamically creating the correct type of token based on scope.

This is currently based off of [mikemountain-apptoken-repo-create-global](https://github.com/hashicorp/boundary/pull/6328) and will need to be rebased once that is merged.

## PCI review checklist

- [ ] I have documented a clear reason for, and description of, the change I am making.
- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.
- [ ] If applicable, I've documented the impact of any changes to security controls.
  Examples of changes to security controls include using new access control methods, adding or removing logging pipelines, etc.

### CODE REVIEW

File: internal/apptoken/repository.go [hunk 1]
```diff
@@ -97,29 +96,82 @@ func (r *Repository) CreateAppToken(ctx context.Context, token *AppToken) (*AppT
...
+		ExpirationTime:            token.ExpirationTime,
+		Revoked:                   token.Revoked,
+		TimeToStaleSeconds:        token.TimeToStaleSeconds,
+		Permissions:               token.Permissions,
+		Token:                     cipherToken,
+	}
+	return newAppToken, nil
+}
+
+func (r *Repository) createAppTokenGlobal(ctx context.Context, token *AppToken, publicId string, dbInserts []interface{}) (*AppToken, []interface{}, error) {
```
Comment (bosorawis) [inline internal/apptoken/repository.go]: Could you remove `dbInserts` from the parameter? It seems like it's both mutating the input AND returning the mutated input.

File: internal/apptoken/repository.go [hunk 2]
```diff
@@ -191,44 +242,105 @@ func (r *Repository) CreateAppToken(ctx context.Context, token *AppToken) (*AppT
...
-			for _, appTokenObject := range dbInserts {
-				if err := w.Create(ctx, appTokenObject); err != nil {
-					return err
-				}
-			}
-			return nil
+	return token, dbInserts, nil
+}
+
+func (r *Repository) createAppTokenOrg(ctx context.Context, token *AppToken, publicId string, dbInserts []interface{}) (*AppToken, []interface{}, error) {
```
Comment (bosorawis) [inline internal/apptoken/repository.go]: same comment on `dbInserts` parameter

File: internal/apptoken/testing.go [hunk 3]
```diff
@@ -182,7 +182,7 @@ func tempTestAddGrants(t *testing.T, repo *Repository, tokenId, scopeId string,
 }
 
 // these will eventually expand to cover org and proj
-func testCheckPermissionGlobal(t *testing.T, repo *Repository, appTokenId string, wantPerms []testPermission) error {
+func testCheckPermission(t *testing.T, repo *Repository, appTokenId string, scopeId string, wantPerms []testPermission) error {
```
Comment (AprilMay0) [inline internal/apptoken/testing.go:185]: I think the `permQuery` below needs to be updated to include org. There are failing tests in `TestRepository_CreateAppToken` I think due to missing permissions.

File: internal/apptoken/repository_test.go [hunk 4]
```diff
@@ -384,6 +551,28 @@ func TestRepository_CreateAppToken(t *testing.T) {
 			wantIsError: 100,
 			wantErrMsg:  "only one of descendants or children grant scope can be specified",
 		},
+		{
+			name: "invalid-org-not-child",
```
Comment (bosorawis) [inline internal/apptoken/repository_test.go:556]: add a test to check that `children` and individual projects aren't allowed?

File: internal/apptoken/repository.go [hunk 5]
```diff
@@ -259,3 +352,37 @@ func (r *Repository) getAppTokenById(ctx context.Context, id string) (*AppToken,
 	}
 	return &at, nil
 }
+
+// processPermissionGrants validates grants and creates grant objects for insertion
+func (r *Repository) processPermissionGrants(ctx context.Context, permId string, grants []string) ([]interface{}, error) {
```
Comment (bosorawis) [inline internal/apptoken/repository.go]: This doesn't need to be a method
```suggestion
func processPermissionGrants(ctx context.Context, permId string, grants []string) ([]interface{}, error) {
```

File: internal/apptoken/repository_test.go [hunk 6]
```diff
@@ -384,6 +552,45 @@ func TestRepository_CreateAppToken(t *testing.T) {
...
+						GrantedScopes: []string{"this", proj2.GetPublicId()},
+					},
+				},
+			},
+			wantErr:     true,
+			wantIsError: errors.Exception,
+			wantErrMsg:  "is not a child of org",
+		},
+		{
+			name: "invalid-org-project-and-children",
```
Comment (bosorawis) [inline internal/apptoken/repository_test.go:578]: add grant_scope `descendants` test for negative case
